### ISSUE TITLE
Stagger issue on with appear-visible

### ISSUE THREAD

**Issue by 7iomka** [ingram-projects/animxyz#70]
Hi.
I have wrong behavior when I use stagger with appear-visible.
Stagger make delay between each children, but if parent (in my case `<XyzTransitionGroup />` (in Vue)) have appear-visible,
that behavior needed changes: delay needed to be calculated between visible items, not total items!
Why?
If you have 15 items, 3 items per row, and for example from the docs
```
<XyzTransitionGroup
          appear-visible
          xyz="fade front-3 flip-down-50% duration-10 stagger-5"
         class="row"
>
   <div
            v-for="(item, i) in items"
            :key="i"
   > ...
   </div>
</XyzTransitionGroup>
```
First 3 items appear normally,
second 3 items - ok.
But when user scroll page, next 3 items and other already have a **_noticeable delay in their general appearance_** (the delay between each other still corresponds to the one specified in the settings).

How to fix that?
Because of this, stagger for me is unusable for that case (is very common!).

**Comment by xxSkyy** [comment]
I've got exact same problem. It'd be good if it'd be fixed or added stagger-all for group or something

**Comment by 7iomka** [comment]
Expected behavior for stagger animation with appear-visible parent
https://codepen.io/GreenSock/pen/823312ec3785be7b25315ec2efd517d8

**Comment by mattaningram** [comment]
Yeah this is something that will take some changes to the logic to fix. Potentially we can automatically update the `--xyz-index` variable as the elements come into view, but I wouldn't be surprised if that causes glitchy behavior if you scroll very quickly. We'll take a look.

In the mean time I would suggest perhaps removing the stagger overall or manually adding a stagger to the first few elements with `--xyz-index` and then the rest just appear normally as they become visible.

**Comment by milesingrams** [comment]
This is definitely not doable with pure CSS. Will look into a javascript approach but ultimately the core idea of this project was to have a CSS based approach with framework specific wrappers to toggle the classes/variables. The main issue is that when using `:nth-child` for delays they stack up leading to big delays for later elements. Since CSS does not allow `:nth-of-type` to target classes, we have no CSS way of detecting, say, the third element **Currently** animating and instead can only detect its index position in the parent.
